docs/research/lcm.md

```diff
@@ -0,0 +1,29 @@
+# AppController Mysql Multi Slave research
+
+This research is meant to provide a guidance on how to achieve ability of AppController-deployed Mysql Multi Slave cluster to cope with losing the master node by promoting one of slave nodes to master.
+This is an example which will allow us to use AC for complicated lcm cases.
+
+MySQL Masters should not be restarted. If the master node crashes, the pod eligible for promotion is transformed into master. Each pod eligible for promotion should have an executable which will perform it's promotion.
+
+AC will need additional process traversing the deployment graph and checking the state of vertices. If the vertex is not ready and has deployed children, we need to take action, which will be defined in an annotation in the pod. We need a DSL for defining these actions, which will allow AC to read specification of these action and run them. This could be the implementation of `failurePolicy` from failure-handling research.
+
+Suggested annotation format:
+```
+{
+  "onFail": [
+    {"type": "exec", "cmd": "promote_to_master.sh", "oneOf": ["slave1, slave2, slave3"]}, # promote one of slaves to master
+    {"type": "create", "template": "slave pod template"} # create new slave pod which will replicate new master
+  ]
```

Comment:
network partition problem is must have there, we need to solve it there. The solution *shouldnt* allow for multiple masters.

On master failure we need to:
- select new master
- be sure that when "current" master is back online it will be discarded.
And no we cannot rely on k8s mechanisms for that because of exactly the problem that @dshulyak mentioned

What we're also missing: 
- we need to be able to connect to mysql, therefore we need at least 2 k8s services, one for master one for slaves, it needs to be updated too, it's not described there at all.